add_on_usage.go

```diff
@@ -0,0 +1,126 @@
+package recurly
+
+import (
+	"context"
+	"encoding/xml"
+	"fmt"
+	"net/http"
+)
+
+// AddOnUsageService manages the interactions for add-on usages.
+type AddOnUsageService interface {
+	// List returns a pager to paginate usages for add-on in subscription. PagerOptions are used to
+	// optionally filter the results.
+	//
+	// https://dev.recurly.com/docs/list-add-ons-usage
+	List(subUUID, addOnCode string, opts *PagerOptions) Pager
+
+	// Create creates a new usage for add-on in subscription.
+	//
+	// https://dev.recurly.com/docs/log-usage
+	Create(ctx context.Context, subUUID, addOnCode string, usage AddOnUsage) (*AddOnUsage, error)
+
+	// Get retrieves an usage. If the usage does not exist,
+	// a nil usage and nil error are returned.
+	//
+	// https://dev.recurly.com/docs/lookup-usage-record
+	Get(ctx context.Context, subUUID, addOnCode, usageId string) (*AddOnUsage, error)
+
+	// Update updates the usage information. Once usage is billed, only MerchantTag can be updated.
+	//
+	// https://dev.recurly.com/docs/update-usage
+	Update(ctx context.Context, subUUID, addOnCode, usageId string, usage AddOnUsage) (*AddOnUsage, error)
+
+	// Delete removes an usage from subscription add-on. If usage is billed, it can't be removed
+	//
+	// https://dev.recurly.com/docs/delete-a-usage-record
+	Delete(ctx context.Context, subUUID, addOnCode, usageId string) error
+}
+
+// Usage is a billable event or group of events recorded on a purchased usage-based add-on and billed in arrears each billing cycle.
+//
+// https://dev.recurly.com/docs/usage-record-object
+type AddOnUsage struct {
+	XMLName            xml.Name `xml:"usage"`
+	Id                 int      `xml:"id,omitempty"`
```

Comment:
Field should be `ID` (See [initialisms](https://github.com/golang/go/wiki/CodeReviewComments#initialisms))

Reply:
Of course, I'm so sorry I miss that - changed